src/web/components/SideBar.tsx

```diff
@@ -0,0 +1,156 @@
+import { useState } from 'react';
+import { Link } from 'react-router-dom';
+import {
+  makeStyles,
+  Button,
+  tokens,
+} from "@fluentui/react-components";
+import {
+  Home24Regular,
+  Home24Filled,
+  DataUsage24Regular,
+  DataUsage24Filled,
+  BookOpen24Regular,
+  BookOpen24Filled,
+  Code24Regular,
+  Code24Filled,
+  Database24Regular,
+  Database24Filled,
+  Cloud24Regular,
+  Cloud24Filled,
+  LearningApp24Regular,
+  LearningApp24Filled,
+  People24Regular,
+  People24Filled,
+  PanelLeftExpandRegular,
+  PanelLeftContractRegular,
+} from "@fluentui/react-icons";
+
+// Styles for the sidebar using Fluent UI's `makeStyles`
+const useStyles = makeStyles({
+    sidebar: {
+      minHeight: '100vh',
+      backgroundColor: '#f4f6f8',
+      display: 'flex',
+      flexDirection: 'column',
+      position: 'relative',
+      transition: 'width 0.2s ease',
+      paddingTop: '40px',
+      paddingRight: '18px',
+      paddingLeft: '16px',
+      paddingBottom: '16px',
+      marginTop: '0',
+    },
+    collapsed: {
+      width: '80px',
+    },
+    expanded: {
+      width: '216px',
+    },
+    toggleButton: {
+      position: 'absolute',
+      top: '10px',
+      right: '8px',
+    },
+    menuItem: {
+      display: 'flex',
+      alignItems: 'center',
+      padding: '10px 12px',
+      color: '#333',
+      textDecoration: 'none',
+      borderRadius: '6px',
+      position: 'relative',
+      transition: 'background-color 0.2s ease, color 0.2s ease',
+      '&:hover': {
+        backgroundColor: '#E6E6E6', // Updated neutral grey background on hover
```

Comment:
Lets not add comments for CSS, most are self explanitory, also colors should be variables. Also lets think of theming early

Reply:
allright